pilot/pkg/serviceregistry/kube/cluster.go

```diff
@@ -0,0 +1,26 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/kube"
+)
+
+// Cluster extends the model.Cluster interface to provide a k8s Client.
```

Comment:
I'm pretty worried by the use of Cluster - which conflicts with envoy cluster. 

How about K8SCluster - and do we really need the model object as well ? How about keeping this local here or in the multicluster controller. 

I don't think we have any plan on implementing k8s MCS on non-k8s environments.

Reply:
This structure is already scoped by `kube` and we already have the notion of `ClusterID` throughout. I don't think that attempting to prefix it with `k8s` would be helpful.